src/main/java/org/itsallcode/junit/sysextensions/security/ExitGuardSecurityManager.java

```diff
@@ -25,6 +28,12 @@
 public class ExitGuardSecurityManager extends SecurityManager
 {
     private boolean trapExit = false;
+    private final SecurityManager delegate;
+
+    public ExitGuardSecurityManager(final SecurityManager delegate)
+    {
+        this.delegate = delegate;
```

Comment:
Check if argument is `null` and use a default SecurityManager with empty methods. Then you can avoid the `null` checks when delegating.

Reply:
I thought about this, but I find the other option less invasive.